### PR TITLE
[docs] Add page on Multi-factor Authentication

### PR DESCRIPTION
**Pull request by jamesorlakin**
Whilst debating enforcing MFA within our team, I realised there isn't a lot of context to the side effects of enabling it. Most of us use Git over HTTP and would need to add a token.

I plan to add another PR that adds a sentence to the UI about needing to generate a token when enabling MFA if HTTP is to be used.

### CODE REVIEW

**Comment by jolheiser** [comment]
The API can still be used with 2FA by setting the `X-Gitea-OTP` header.

The docs mention this in [API Usage](https://docs.gitea.io/en-us/api-usage/#using-basic-authentication).

**Comment by jamesorlakin** [comment]
Aah yes, I didn't spot that! I guess this still applies for the Git CLI though?

**Comment by jolheiser** [comment]
Yes, currently over HTTP/S you would need an app token (or SSH).

File: docs/content/doc/usage/multi-factor-authentication.en-us.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Enabling MFA on a user does affect how the Git HTTP protocol can be used with the Git CLI.
+This interface does not support MFA, and trying to use a password normally will no longer be possible whilst MFA is enabled.
+If SSH is not an option for Git operations, an access token can be generated within the "Applications" tab of the user settings page.
+This access token can be used as if it were a password in order to allow the Git CLI to function over HTTP.
+
+> **Warning** - By its very nature, an access token sidesteps the security benefits of MFA.
+> It must be kept secure and should only be used as a last resort.
+
+The Gitea API supports providing the relevant TOTP password in the `X-Gitea-OTP` header, as described in [API Usage](https://docs.gitea.io/en-us/api-usage).
+This should be used instead of an access token where possible.
```
Comment (6543) [inline docs/content/doc/usage/multi-factor-authentication.en-us.md:35]: as api is for automation ... it's more secure to use a token than basic-auth + TOTP!

you only have to make sure to name access token on creation properly and only use it for one thing! (CI, GitNex, ...)

Comment (6543) [reply docs/content/doc/usage/multi-factor-authentication.en-us.md:35]: so you can precicely add and remove access application level based ...

File: docs/content/doc/usage/multi-factor-authentication.en-us.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+Multi-factor Authentication (also referred to as MFA or 2FA) enhances security by requiring a time-sensitive set of credentials in addition to a password.
+If a password were later to be compromised, Gitea would still not allow a successful login and the account would remain secure.
+Gitea supports both TOTP (Time-based One-Time Password) tokens and FIDO-based hardware keys.
+
+MFA can be configured within the "Security" tab of the user settings page.
+
+## MFA Considerations
+
+Enabling MFA on a user does affect how the Git HTTP protocol can be used with the Git CLI.
+This interface does not support MFA, and trying to use a password normally will no longer be possible whilst MFA is enabled.
```
Comment (bagasme) [inline docs/content/doc/usage/multi-factor-authentication.en-us.md:27]: did you mean Git CLI doesn't support MFA?

**Comment by denyskon** [comment]
@jamesorlakin Are you still interested in this PR? If so, could you update it, as we recently changed our documentation system? This would involve a new file location and some changes inside frontmatter.

**Comment by lunny** [comment]
Some contents are outdated, maybe we need to create a new PR or update this one.

**Comment by lunny** [comment]
replaced by #26654

**Pull request closed without merging**